specs/experimental/custom-gas-token.md

```diff
@@ -0,0 +1,488 @@
+symbol MUST be less than or equal to 32 bytes. If the token passes all of these checks,\
+`OptimismPortal.setGasPayingToken` is called. The implementation of `initialize` MUST not allow the chain
+operator to change the address of the custom gas token if it is already set.
+
+#### Getters
+
+Some getters are added to the `SystemConfig`:
+
+```solidity
+function gasPayingToken() public view returns (address, uint8);
+function gasPayingTokenName() public view returns (string);
+function gasPayingTokenSymbol() public view returns (string);
+```
+
+If custom gas token is not used, then `gasPayingToken()` should return `(ETHER_TOKEN_ADDRESS,18)`,
+`gasPayingTokenName` should return `Ether` and `gasPayingTokenSymbol` should return `ETH`.
+
+### L1Block
+
+The `L1Block` contract is upgraded with a permissioned function `setGasPayingToken` that is used to set
+information about the gas paying token. The `DEPOSITOR_ACCOUNT` MUST be the only account that can call the
+setter function. This setter is differentiated from the `setL1BlockValues` functions because the gas paying
+token is not meant to be dynamic config whereas the arguments to `setL1BlockValues` are generally dynamic in nature.
+
+Any L2 contract that wants to learn the address of the gas paying token can call the getter on the
+`L1Block` contract.
+
+```solidity
+function setGasPayingToken(address _token, uint8 _decimals, byte32 _name, bytes32 _symbol) external;
+function gasPayingToken() public view returns (address, uint8);
+function gasPayingTokenName() public view returns (string);
+function gasPayingTokenSymbol() public view returns (string);
+```
+
+If custom gas token is not used, then `gasPayingToken()` should return `(ETHER_TOKEN_ADDRESS,18)`,
+`gasPayingTokenName` should return `Ether` and `gasPayingTokenSymbol` should return `ETH`.
+
+### WETH9
+
+The `WETH9` predeploy is updated so that it calls out to the `L1Block` contract to retreive the `name` and `symbol`.
+This allows for front ends to more easily trust the token contract. It should prepend `Wrapped` to the `name`
+and `W` to the `symbol`.
+
+## User Flow
+
+The user flow for custom gas token chains is slightly different than for chains that use `ether`
+to pay for gas. The following tables highlight the methods that can be used for depositing and
+withdrawing the native asset. Not every interface is included.
+
+The `StandardBridge` and `CrossDomainMessenger` are symmetric in their APIs between L1 and L2
+meaning that "sending to the other domain" indicates it can be used for both deposits and withdrawals.
+
+### When ETH is the Native Asset
+
+| Scenario | Method | Prerequisites |
+|----------|--------|---------------|
+| Native Asset Send to Other Domain | `L1StandardBridge.bridgeETH(uint32,bytes) payable` | None |
+| Native Asset Send to Other Domain | `L1CrossDomainMessenger.sendMessage(address,bytes,uint32) payable` | None |
```

Comment:
Should the second row's scenario be "Native Asset and/or Message Send to Other Domain"? Something to distinguish it's use case from the prior row